File: frontend/src/types/field.ts
export interface Field {
  npdid_field: number;
  name: string;
  main_area: string | null;
  status: string | null;
  hc_type: string | null;
  operator: string | null;
  discovery_year: number | null;
  lat: number | null;
  lon: number | null;
  co2_mol_pct: number | null;
  co2_spec?: CO2Spec | null;
}

export interface Discovery {
  npdid_discovery: number;
  name: string;
  main_area: string;
  status: string;
  hc_type: string;
  operator: string;
  discovery_year: number | null;
  lat: number | null;
  lon: number | null;
  co2_spec?: CO2Spec | null;
}

export interface CO2Spec {
  co2_mol_pct: number;
  co2_mol_pct_range_low: number | null;
  co2_mol_pct_range_high: number | null;
  source: string;
}

File: backend/app/models/field.py
"""Field model -- represents an NCS producing field."""

from typing import Optional

from sqlalchemy import Column, Integer, String, Float

from app.database import Base


class Field(Base):
    __tablename__ = "fields"

    npdid_field = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False, index=True)
    main_area = Column(String, nullable=True)
    status = Column(String, nullable=True)
    hc_type = Column(String, nullable=True)
    operator = Column(String, nullable=True)
    discovery_year = Column(Integer, nullable=True)
    lat = Column(Float, nullable=True)
    lon = Column(Float, nullable=True)
